lib/topic_view.rb

```diff
@@ -73,8 +73,7 @@ def self.allowed_post_custom_fields(user, topic)
   end
 
   def self.add_custom_filter(key, &blk)
-    @custom_filters ||= {}
-    @custom_filters[key] = blk
+    (@custom_filters ||= {})[key] = blk
```

Comment:
Same remark regarding defining `@custom_filters` in `.custom_filters` :slightly_smiling_face: